daffodil-udf/src/test/scala/org/sbadudfs/udfexceptions/StringFunctions/Reverse.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sbadudfs.udfexceptions.StringFunctions
+
+import org.apache.daffodil.udf.UserDefinedFunction
+import org.apache.daffodil.udf.UserDefinedFunctionIdentification
+import com.sun.xml.internal.bind.v2.runtime.IllegalAnnotationsException
+
+/**
+ * UDF for exceptions negative tests in Scala
+ *
+ * Throws exception on evaluate
+ */
+@UserDefinedFunctionIdentification(name = "reverse",
+  namespaceURI = "http://scala.udf.com")
+class Reverse extends UserDefinedFunction {
+  final case class CustomException(
```

Comment:
Looks like this final case class is causing build warnings/failures. Not sure what the problem is, but the solution might be to just move it outside of the UDF, or define it in the provider and share it among these bad UDFS?